types/field_type.go

```diff
@@ -40,6 +40,15 @@ func NewFieldType(tp byte) *FieldType {
 	}
 }
 
+// CloneFieldType clones the given FieldType.
+func CloneFieldType(src *FieldType) *FieldType {
```

Comment:
Please clone all the attributes in a FieldType.

Reply:
If you just want a shallow copy, the following will be easier:
```go
ft := *src
return &ft
```

Reply:
@lamxTyler Thank you for the tip!

Reply:
Why not let it be a member function of `FieldType`?

Reply:
@winoros Yes, we can further refactor the function by making it to be the member function in ``parser/types/field_type.go``.  I'll handle it another PR.